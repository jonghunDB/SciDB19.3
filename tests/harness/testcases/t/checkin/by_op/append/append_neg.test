--setup
create array TWO_D<v:double>[x=0:9; t=0:*]
store(build(<v:double>[x=0:9; t=0:19], float(t)+float(x)), STUFF)

--test

--echo Some of stuff...
op_sum(STUFF, v)

# Needs a dimension name to work.
--error --code=scidb::SCIDB_SE_INFER_SCHEMA::SCIDB_LE_MULTIDIMENSIONAL_ARRAY_NOT_ALLOWED \
    "append(STUFF, TWO_D)"
--igdata "append(STUFF, TWO_D, t)"
# Append worked, because sum is same as for STUFF.
op_sum(TWO_D, v)

# Dimension name is valid, but data won't fit.
--error --code=scidb::SCIDB_SE_EXECUTION::SCIDB_LE_CHUNK_OUT_OF_BOUNDARIES \
    "append(STUFF, TWO_D, x)"

# Append to non-existent array.
--error --code=scidb::SCIDB_SE_QPROC::SCIDB_LE_ARRAY_DOESNT_EXIST \
    "append(build(STUFF, random()), IDONTEXIST)"

# Append to non-existent dataframe.
--error --code=scidb::SCIDB_SE_QPROC::SCIDB_LE_ARRAY_DOESNT_EXIST \
    "append(flatten(build(STUFF, random())), IDONTEXIST)"

--cleanup
remove(STUFF)
remove(TWO_D)
