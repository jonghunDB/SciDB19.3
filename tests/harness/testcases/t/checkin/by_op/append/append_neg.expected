Query was executed successfully

Query was executed successfully

Some of stuff...
{i} v_sum
{0} 2800

[An error expected at this place for the query "append(STUFF, TWO_D)". And it failed with error code = scidb::SCIDB_SE_INFER_SCHEMA::SCIDB_LE_MULTIDIMENSIONAL_ARRAY_NOT_ALLOWED. Expected error code = scidb::SCIDB_SE_INFER_SCHEMA::SCIDB_LE_MULTIDIMENSIONAL_ARRAY_NOT_ALLOWED.]

[Query was executed successfully, ignoring data output by this query.]

{i} v_sum
{0} 2800

[An error expected at this place for the query "append(STUFF, TWO_D, x)". And it failed with error code = scidb::SCIDB_SE_EXECUTION::SCIDB_LE_CHUNK_OUT_OF_BOUNDARIES. Expected error code = scidb::SCIDB_SE_EXECUTION::SCIDB_LE_CHUNK_OUT_OF_BOUNDARIES.]

[An error expected at this place for the query "append(build(STUFF, random()), IDONTEXIST)". And it failed with error code = scidb::SCIDB_SE_QPROC::SCIDB_LE_ARRAY_DOESNT_EXIST. Expected error code = scidb::SCIDB_SE_QPROC::SCIDB_LE_ARRAY_DOESNT_EXIST.]

[An error expected at this place for the query "append(flatten(build(STUFF, random())), IDONTEXIST)". And it failed with error code = scidb::SCIDB_SE_QPROC::SCIDB_LE_ARRAY_DOESNT_EXIST. Expected error code = scidb::SCIDB_SE_QPROC::SCIDB_LE_ARRAY_DOESNT_EXIST.]

Query was executed successfully

Query was executed successfully

